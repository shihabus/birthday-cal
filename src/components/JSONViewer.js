import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { fromEpoch } from "../utils/dateUtils";

const Container = styled.div`
  grid-area: viewer;
  height: 100%;
  overflow: auto;
  color: #6a6a6a;
  background-color: white;
  padding: 20px 10px;
`;

const ListItem = styled.pre`
  margin-left: 40px;
  color: #000;
  color: #6a6a6a;
`;

export default function JSONViewer(props) {
  return (
    <Container>
      {"["}
      {[
        props.data.map(person => (
          <section
            data-id={person.id}
            key={person.id}
            style={{ fontSize: "14px" }}
          >
            <pre> {" {"}</pre>
            <ListItem> name : {person.name}</ListItem>
            <ListItem> birthday : {fromEpoch(person.birthday)}</ListItem>
            <pre> {" },"}</pre>
          </section>
        ))
      ]}
      {"]"}
    </Container>
  );
}

JSONViewer.propTypes = {
  data: PropTypes.arrayOf(PropTypes.object)
};
