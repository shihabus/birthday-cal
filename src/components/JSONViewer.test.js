import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import JSONViewer from "./JSONViewer";

test("JSONViewer snapshot", () => {
  const mockData = [];
  const { container } = render(<JSONViewer data={mockData} />);
  expect(container.firstChild).toMatchSnapshot();
});
