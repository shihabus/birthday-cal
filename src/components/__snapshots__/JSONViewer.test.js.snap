// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSONViewer snapshot 1`] = `
<div
  class="sc-AxjAm kwluUN"
>
  [
  ]
</div>
`;
